### PR TITLE
resource/aws_route53_resolver_firewall_rule_group: Add new resource

### PR DESCRIPTION
**Pull request by gazoakley**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #18520

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSRoute53ResolverFirewallRuleGroup_'                       
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSRoute53ResolverFirewallRuleGroup_ -timeout 180m
=== RUN   TestAccAWSRoute53ResolverFirewallRuleGroup_basic
=== PAUSE TestAccAWSRoute53ResolverFirewallRuleGroup_basic
=== RUN   TestAccAWSRoute53ResolverFirewallRuleGroup_disappears
=== PAUSE TestAccAWSRoute53ResolverFirewallRuleGroup_disappears
=== RUN   TestAccAWSRoute53ResolverFirewallRuleGroup_tags
=== PAUSE TestAccAWSRoute53ResolverFirewallRuleGroup_tags
=== CONT  TestAccAWSRoute53ResolverFirewallRuleGroup_basic
=== CONT  TestAccAWSRoute53ResolverFirewallRuleGroup_tags
=== CONT  TestAccAWSRoute53ResolverFirewallRuleGroup_disappears
--- PASS: TestAccAWSRoute53ResolverFirewallRuleGroup_disappears (18.43s)
--- PASS: TestAccAWSRoute53ResolverFirewallRuleGroup_basic (25.56s)
--- PASS: TestAccAWSRoute53ResolverFirewallRuleGroup_tags (61.38s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	63.040s
```

### CODE REVIEW

**Comment by gazoakley** [comment]
This currently branches from #18544 - if a later build deps bump gets merged this can be rebased

File: website/docs/r/route53_resolver_firewall_rule_group.markdown [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+* `name` - (Required) A name that lets you identify the rule group, to manage and use it.
+* `tags` - (Optional) A map of tags to assign to the resource.
+
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `arn` - The ARN (Amazon Resource Name) of the rule group.
+* `id` - The ID of the rule group.
```
Comment (ewbankkit) [inline website/docs/r/route53_resolver_firewall_rule_group.markdown:33]: Add `owner_id` and `share_status` exported attributes.

File: aws/resource_aws_route53_resolver_firewall_rule_group_test.go [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+	}
+	conn := client.(*AWSClient).route53resolverconn
+	var sweeperErrs *multierror.Error
+
+	err = conn.ListFirewallRuleGroupsPages(&route53resolver.ListFirewallRuleGroupsInput{}, func(page *route53resolver.ListFirewallRuleGroupsOutput, isLast bool) bool {
+		if page == nil {
+			return !isLast
+		}
+
+		for _, queryLogConfig := range page.FirewallRuleGroups {
```
Comment (ewbankkit) [inline aws/resource_aws_route53_resolver_firewall_rule_group_test.go:40]: `queryLogConfig` -> `firewallRuleGroup`

File: aws/resource_aws_route53_resolver_firewall_rule_group.go [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+	return nil
+}
+
+func resourceAwsRoute53ResolverFirewallRuleGroupUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).route53resolverconn
+
+	if d.HasChange("tags") {
+		o, n := d.GetChange("tags")
+		if err := keyvaluetags.Route53resolverUpdateTags(conn, d.Get("arn").(string), o, n); err != nil {
+			return fmt.Errorf("error updating Route53 Resolver DNS Firewall rule group (%s) tags: %s", d.Get("arn").(string), err)
```
Comment (ewbankkit) [inline aws/resource_aws_route53_resolver_firewall_rule_group.go:127]:
```suggestion
			return fmt.Errorf("error updating Route53 Resolver DNS Firewall rule group (%s) tags: %w", d.Get("arn").(string), err)
```

**Comment by gazoakley** [comment]
Thanks @ewbankkit - I've pushed a commit addressing the comments

**Comment by ewbankkit** [comment]
May need rebasing after #18544 (or subsequent) is merged.

**Review by ewbankkit — APPROVED** [review]
LGTM.

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSRoute53ResolverFirewallRuleGroup_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSRoute53ResolverFirewallRuleGroup_ -timeout 180m
go: downloading github.com/aws/aws-sdk-go v1.38.11
=== RUN   TestAccAWSRoute53ResolverFirewallRuleGroup_basic
=== PAUSE TestAccAWSRoute53ResolverFirewallRuleGroup_basic
=== RUN   TestAccAWSRoute53ResolverFirewallRuleGroup_disappears
=== PAUSE TestAccAWSRoute53ResolverFirewallRuleGroup_disappears
=== RUN   TestAccAWSRoute53ResolverFirewallRuleGroup_tags
=== PAUSE TestAccAWSRoute53ResolverFirewallRuleGroup_tags
=== CONT  TestAccAWSRoute53ResolverFirewallRuleGroup_basic
=== CONT  TestAccAWSRoute53ResolverFirewallRuleGroup_tags
=== CONT  TestAccAWSRoute53ResolverFirewallRuleGroup_disappears
--- PASS: TestAccAWSRoute53ResolverFirewallRuleGroup_disappears (14.32s)
--- PASS: TestAccAWSRoute53ResolverFirewallRuleGroup_basic (19.06s)
--- PASS: TestAccAWSRoute53ResolverFirewallRuleGroup_tags (40.29s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	40.420s
```

**Comment by breathingdust** [comment]
LGTM :rocket: Thanks @gazoakley!

Verified Acceptance Tests in Commercial (us-west-2)

```console
make testacc TESTARGS='-run=TestAccAWSRoute53ResolverFirewallRuleGroup_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSRoute53ResolverFirewallRuleGroup_ -timeout 180m
go: downloading github.com/aws/aws-sdk-go v1.38.12
go: downloading github.com/hashicorp/go-version v1.3.0
=== RUN   TestAccAWSRoute53ResolverFirewallRuleGroup_basic
=== PAUSE TestAccAWSRoute53ResolverFirewallRuleGroup_basic
=== RUN   TestAccAWSRoute53ResolverFirewallRuleGroup_disappears
=== PAUSE TestAccAWSRoute53ResolverFirewallRuleGroup_disappears
=== RUN   TestAccAWSRoute53ResolverFirewallRuleGroup_tags
=== PAUSE TestAccAWSRoute53ResolverFirewallRuleGroup_tags
=== CONT  TestAccAWSRoute53ResolverFirewallRuleGroup_basic
=== CONT  TestAccAWSRoute53ResolverFirewallRuleGroup_tags
=== CONT  TestAccAWSRoute53ResolverFirewallRuleGroup_disappears
--- PASS: TestAccAWSRoute53ResolverFirewallRuleGroup_disappears (14.19s)
--- PASS: TestAccAWSRoute53ResolverFirewallRuleGroup_basic (17.62s)
--- PASS: TestAccAWSRoute53ResolverFirewallRuleGroup_tags (38.94s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	41.982s
```

Verified Acceptance Tests in GovCloud (us-gov-west-1)

```console
make testacc TESTARGS='-run=TestAccAWSRoute53ResolverFirewallRuleGroup_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSRoute53ResolverFirewallRuleGroup_ -timeout 180m
=== RUN   TestAccAWSRoute53ResolverFirewallRuleGroup_basic
=== PAUSE TestAccAWSRoute53ResolverFirewallRuleGroup_basic
=== RUN   TestAccAWSRoute53ResolverFirewallRuleGroup_disappears
=== PAUSE TestAccAWSRoute53ResolverFirewallRuleGroup_disappears
=== RUN   TestAccAWSRoute53ResolverFirewallRuleGroup_tags
=== PAUSE TestAccAWSRoute53ResolverFirewallRuleGroup_tags
=== CONT  TestAccAWSRoute53ResolverFirewallRuleGroup_basic
=== CONT  TestAccAWSRoute53ResolverFirewallRuleGroup_tags
=== CONT  TestAccAWSRoute53ResolverFirewallRuleGroup_disappears
--- PASS: TestAccAWSRoute53ResolverFirewallRuleGroup_disappears (7.90s)
--- PASS: TestAccAWSRoute53ResolverFirewallRuleGroup_basic (10.56s)
--- PASS: TestAccAWSRoute53ResolverFirewallRuleGroup_tags (24.39s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	27.241s
```

**Pull request merged**